Modules/QuestieValidateGameCache.lua

```diff
@@ -0,0 +1,145 @@
+        return
+    end
+
+    --[[
+    -- Player can have 0 quests in questlog OR game's cache can have empty questlog while cache is invalid
+    -- This is a workaround to wait and see if player really has 0 quests.
+    -- Without cached information the first QLU does not have any quest log entries.
+    -- After MAX_INIT_QUEST_LOG_TRIES tries we stop trying
+    if numEntries == 0 then
+        initQuestLogTries = initQuestLogTries + 1
+        if initQuestLogTries < MAX_INIT_QUEST_LOG_TRIES then
+            print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is empty. Tries:", initQuestLogTries.."/"..MAX_INIT_QUEST_LOG_TRIES)
+            _WaitForGameCache_TryAgainAtEvent()
+            return
+        else
+            print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Decide player has no quests for real.")
+        end
+    end
+    ]]--
+
+    local isGameCacheGood = true
+    local goodQuestsCount = 0 -- for debug stats
+
+    for i = 1, MAX_QUEST_LOG_INDEX do
+        local title, _, _, isHeader, _, _, _, questId = GetQuestLogTitle(i)
+        if (not title) then
+            break -- We exceeded the valid quest log entries
+        end
+        if (not isHeader) then
+            local hasInvalidObjective -- for debug stats
+            local objectiveList = GetQuestObjectives(questId)
+            for _, objective in pairs(objectiveList) do -- objectiveList may be {} and that is validly cached quest in game log
+                if (not objective.text) or stringSub(objective.text, 1, 1) == " " then
+                    -- Game hasn't cached the quest fully yet
+                    isGameCacheGood = false
+                    hasInvalidObjective = true
+
+                    -- No early "return false" here to force iterate whole quest log and speed up caching
+                end
+            end
+            if not hasInvalidObjective then
+                goodQuestsCount = goodQuestsCount + 1
+            end
+        end
+    end
+
+    if not isGameCacheGood then
+        print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is not yet okey. Good quest: "..goodQuestsCount.."/"..numQuests)
+        return
+    end
+
+    if goodQuestsCount ~= numQuests then
+        -- This shouldn't be possible
+        Questie:Error("Report this error! Game QuestLog cache is broken. Good quest: "..goodQuestsCount.."/"..numQuests) -- TODO fix message and add translations? translations might not be available?
+        -- TODO should we stop whole addon loading progress?
+    end
+
+    print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is |cff00bc32".."OK".."|r. Good quest: "..goodQuestsCount.."/"..numQuests)
+
+    DestroyEventFrame()
+
+    cacheGood = true
+
+    -- Call all callbacks
+    while (#callbacks > 0) do
+        local callback = table.remove(callbacks, 1)
```

Comment:
Should `table.remove` also get a local variable?

Reply:
Yes, even doesn't really matter as the line is run on average 0 times currently.

Reply:
Done in 78b16dab24437ff7c33ec1111cd51a01e38c7ea2